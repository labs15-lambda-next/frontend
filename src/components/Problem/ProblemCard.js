import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import Tooltip from "@material-ui/core/Tooltip";
import { withStyles } from "@material-ui/styles";

import DefaultImage from '../../static/images/cards/default-image.jpg';

const ImageSetter = require("../../static/stylingComponents/ImageSetter");

const styles = {
  problemCards: { boxShadow: "1.5px silver", borderRadius: "0px" },
  background: { backgroundColor: "#6c757c", width: "100%" },
  callToActionBtn2: {
    backgroundColor: "#bb1333",
    borderRadius: "0px",
    color: "#ffffff",
    "&:hover": {
      backgroundColor: "#750808"
    }
  },
  noTextDecoration: { textDecoration: "none" },
  bolded: { fontWeight: "bold" },
  padded: { paddingBottom: "1rem", paddingRight: "1rem", paddingLeft: "1rem" },
  buttonPadding: { padding: "0.5rem" }
};

const ProblemCard = props => {
  const { problem, classes, signups } = props;

  return (
    <Card className={classes.problemCards} raised>
      <CardMedia
        className={classes.background}
        component="img"
        src={ImageSetter.staticImage(problem.problem_category) || DefaultImage}
        title={problem.problem_title}
      />
      <Grid
        container
        justify="space-between"
        alignItems="flex-end"
        className={classes.backgroundWhite}
      >
        <Grid item>
          <CardContent>
            <Tooltip title={problem.problem_title}>
              <Typography
                className={classes.bolded}
                gutterBottom
                variant="h5"
                color="initial"
                component="h3"
              >
                {problem.problem_title.length > 20
                  ? problem.problem_title.substring(0, 20) + "..."
                  : problem.problem_title}
                :
              </Typography>
            </Tooltip>
            <Typography variant="body2" component="p" gutterBottom>
              {problem.problem_description > 40
                ? problem.problem_description.substring(0, 40) + "..."
                : problem.problem_description}
            </Typography>
            <Typography variant="subtitle2" color="textSecondary" component="p">
              Category: {problem.problem_category}
            </Typography>
          </CardContent>
        </Grid>
        <Grid container alignItems="center" direction="row-reverse">
          <Grid item xs={12} sm={6} className={classes.buttonPadding}>
            <Typography variant="body2" component="p" align="center">
              # of Sign Ups: <span className={classes.bolded}>{signups}</span>
            </Typography>
          </Grid>
          <Grid item xs={12} sm={6}>
            <Link
              to={`/problem-details/${problem.id}`}
              className={classes.noTextDecoration}
            >
              <Button
                className={classes.callToActionBtn2}
                size="medium"
                color="primary"
                fullWidth
              >
                Learn More
              </Button>
            </Link>
          </Grid>
        </Grid>
      </Grid>
    </Card>
  );
};

ProblemCard.defaultProps = {
  problem: {},
  classes: {},
  signups: 0
};

ProblemCard.propTypes = {
  problem: PropTypes.shape({
    id: PropTypes.number,
    problem_category: PropTypes.string,
    problem_title: PropTypes.string,
    problem_description: PropTypes.string
  }),
  classes: PropTypes.objectOf(PropTypes.string),
  signups: PropTypes.number
};

export default withStyles(styles)(ProblemCard);
