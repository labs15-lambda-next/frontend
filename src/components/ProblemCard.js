import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import { styled } from "@material-ui/styles";
import health from "../static/images/cards/health.jpg";
import technology from "../static/images/cards/technology.jpg";
import fitness from "../static/images/cards/fitness1.jpg";
import personal from "../static/images/cards/personal.jpg";
import science from "../static/images/cards/science1.jpg";
import finance from "../static/images/cards/finance.jpg";
import Icon from '@material-ui/core/Icon';

import {
  ProblemCardLink,
  CallToActionBtn2
} from '../static/stylingComponents'

//import PropTypes from 'prop-types';

import { Link as RouterLink  } from "react-router-dom";

const MyCard = styled(Card)({
  
  boxShadow:'2px 4px silver'
});

const MyCardMedia = styled(CardMedia)({
  backgroundColor: "white",
  height:200
});

// const MyCategory = styled(Typography)({
//   fontWeight: "40px"
// });

const category = [
  { name: "health", value: health, icon:'local_hospital'},
  { name: "technology", value: technology, icon:'settings_remote' },
  { name: "fitness", value: fitness, icon:'directions_run' },
  { name: "personal", value: personal, icon:'person' },
  { name: "science", value: science, icon:'eco' },
  { name: "finance", value: finance, icon:'equalizer' }
];


const ProblemCard = props => {
  const problem = props.problems;

  function getImage(image_category) {
    const cat = category.find(
      item => item.name === image_category.toLowerCase()
    );
    return cat.value;
  }

  function getIcon(image_category) {
    const cat = category.find(
      item => item.name === image_category.toLowerCase()
    );
    return cat.icon;
  }

  return (
    <MyCard>
      <ProblemCardLink component={RouterLink} to={`/problem-details/${problem.id}`}>
        <MyCardMedia
          component="img"
          src={getImage(problem.problem_category)}
          title={problem.problem_title}
        />
        <CardContent>
          <Typography gutterBottom variant="headline" color="textSecondary" component="h2">
              {problem.problem_title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
          
              <Icon>{getIcon(problem.problem_category)}</Icon>
              {problem.problem_category}
            
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {problem.problem_description.substring(0, 50) + "..."}
          </Typography>
        </CardContent>
      </ProblemCardLink>
        <CardActions>
          <CallToActionBtn2 size="medium" color="primary" href={`/problem-details/${problem.id}`} >
            Learn More
          </CallToActionBtn2>
        </CardActions>
    </MyCard>
  );
};

export default ProblemCard;
