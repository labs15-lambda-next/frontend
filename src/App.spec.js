import React from 'react';
import renderer from 'react-test-renderer';
import ProblemCard from './components/ProblemCard';

describe('ProblemCard', () => {
  it('ProblemCard Snapshot', () => {
    const problem = {
      id: 1, problem_category: 'technology', problem_title: 'title', problem_description: 'tech problem'
    };

    const component = renderer.create(<ProblemCard problems={problem} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
