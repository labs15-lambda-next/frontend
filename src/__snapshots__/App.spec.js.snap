// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemCard ProblemCard Snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root WithStyles(ForwardRef(Card))-root-1 MuiPaper-rounded"
>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover WithStyles(ForwardRef(Link))-root-30 MuiTypography-colorPrimary"
    onBlur={[Function]}
    onFocus={[Function]}
    to="/problem-details/1"
  >
    <img
      className="WithStyles(ForwardRef(CardMedia))-root-67"
      src="test-file-stub"
      title="title"
    />
    <div
      className="MuiCardContent-root"
    >
      <h3
        className="MuiTypography-root MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
      >
        title
      </h3>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root"
        >
          settings_remote
        </span>
        technology
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        style={
          Object {
            "height": 40,
          }
        }
      >
        tech problem
      </p>
    </div>
  </a>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-80 MuiButton-text MuiButton-textPrimary"
      href="/problem-details/1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Learn More
      </span>
    </a>
  </div>
</div>
`;
